fontir/src/ir.rs

```diff
@@ -102,13 +102,15 @@ impl Features {
 #[serde(from = "GlyphSerdeRepr", into = "GlyphSerdeRepr")]
 pub struct Glyph {
     pub name: GlyphName,
+    pub accessors: HashSet<Vec<u32>>, // sequence(s) of unicodes to reach this glyph
```

Comment:
why don't you call these `unicodes` to match what the sources (e.g. UFO, Glyphs) also have? I personally don't find the name "accessors" any clearer, perhaps it's my non-native English or my familiarity with font sources.

Reply:
if you don't like "unicodes", then "codepoints" would also be fine

Reply:
actually -- I propose that four our IR we completely do away with the flawed approach of storing a list of unicode codepoints at the Glyph level, and instead have a global CharacterMap (which you can store in the static metadata) that is simply a `HashMap<Codepoint, GlyphName>`.
Associating codepoints to a glyph is wrong, not just because it obfuscates the fact that a cmap -- which this piece of data is designed to build -- actually maps from codepoints to glyphs instead of the other way around; but most importantly, it opens the door for ambiguous/invalid/impossible to encode mappings whereby more than one glyph ends up being mapped to the same unicode codepoint (e.g. two glyphs "foo" and "bar" both having unicodes list = [0x0061]; if one types "a", which one will one get, foo or bar?!)

See discussion on the proposal for storing a global cmap.txt and removing GLIF-level unicode elements in UFO4: https://github.com/unified-font-object/ufo-spec/issues/77

Reply:
Since we will support legacy versions of both glyphs and ufo for at least a few years I believe the IR should model it in this way and then the BE should convert it.

I do agree that accessors is a stunningly poor choice of name. I like codepoints and use it in other code.

Reply:
Maybe I will split codepoints from ligatures so it's a bit simpler what goes into cmap vs needs to become ligatures.

Reply:
you mean, nanoemoji-style GSUB ligatures from RGI sequences? Note that the list of codepoints in a UFO/Glyphs glyph is _not_ understood to be a sequence in GSUB ligature sense, but as the set of codepoints that map to that glyph in the character mapping table.

Reply:
it is only a _list_ (ordered) and not a _set_ because of the even more flawed notion of a primary unicode codepoint which happens to be the first in that list (and which makes no sense from a cmap point of view).
The whole glyph.unicodes is a legacy that haunts font tooling since the Robofab days and should really be superseded by a global {codepoint: gid} mapping.

Reply:
Done, it's slightly more verbose but IMHO much clearer

Reply:
I don't want to lose the disucssion of representing cmap some other way, nor do I want to block this on it. Filed as https://github.com/googlefonts/fontmake-rs/issues/145

Reply:
> the list of codepoints in a UFO/Glyphs glyph is not understood to be a sequence in GSUB ligature sense, but as the set of codepoints that map to that glyph in the character mapping table.

I managed to get that right for UFO and wrong for glyphs because _clearly_ a single field unicodes with a list means a ligature. Sigh. Fixed.

I did destroy the ordering, which might be a problem if we ever wanted source => IR => source with retention of primary codepoint. I don't currently understand what a primary codepoint *is*

Reply:
OK, I think thi sis fixed for this PR.